client/state/comments/from-api.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Internal dependencies
+ */
+import { AnonymousUser, WordPressUser } from './models/comment-author';
+import { CommentInfo } from './models/comment-info';
+import { Approved, Pending, Spam, Trash } from './models/comment-status';
+import { LoadedComment } from './models/comment';
+
+const toAuthor = ( { avatar_URL, email, ID, name } ) =>
+	ID > 0
+		? new WordPressUser( ID )
+		: new AnonymousUser( { avatar: avatar_URL, displayName: name, email, url: URL } );
+
+const toStatus = text => {
+	switch ( text ) {
+		case 'approved': return new Approved();
```

Comment:
how is this better than using the text `approved` or an `Approved` flyweight? A new instance for every status seems like overkill.